analyses/optimus/create_synthetic_data.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# this script takes as input the PBMC8k dataset, multi-aligned with Optimus.
+# it generates a bam file containing unique, unaligned, and multialigned reads from different
+# chromosomes
+# it then regenerates fastq data
+
+set -eo pipefail
+
+# grab the picard jar
+curl -o picard.jar https://github.com/broadinstitute/picard/releases/download/2.18.4/picard.jar
+
+# assumes you've grabbed samtools with one of these commands
+# suo apt-get install samtools
+# sudo yum install samtools
+
+INPUT_BAM=$1  # pbmc8k resequenced output
+N_UNIQUE=$2  # 1,000,000
+N_UNALIGNED=$3  # 100,000
+N_MULTIALIGNED=$4  # 100,000
+
+# sort and index the bam file
+samtools sort -o sorted.bam ${INPUT_BAM}
+samtools index sorted.bam
+
+# get some unique-aligned reads
+samtools view -H sorted.bam > header.txt
+samtools view sorted.bam 1 | head -n ${N_UNIQUE} > unique.sam
```

Comment:
do you want to output uniquely mapped reads or primary alignment? if the second, you can use -f16 to extract primary alignment.

Reply:
Here I want primary. However, I could use this for the multi-aligned reads below where I want a read that aligns more than once, but I want the primary read only for the synthetic data.